src/args.c

```diff
@@ -1216,3 +1267,281 @@ void test_dnsmasq_options(int argc, const char *argv[])
+			}
+		}
+		else if(argc == 6)
+		{
+			// pihole-FTL --config <some key> ...
+			for(unsigned int i = 0; i < CONFIG_ELEMENTS; i++)
+			{
+				struct conf_item *conf_item = get_conf_item(&config, i);
+				if(!conf_item)
+					continue;
+				if(strcmp(conf_item->k, argv[4]) == 0)
+				{
+					// See if we can suggest a value
+					if(conf_item->t == CONF_BOOL || conf_item->t == CONF_ALL_DEBUG_BOOL)
+					{
+						// pihole-FTL --config <boolean option>> ...
+						const char *options[] = {
+							"true", "false"
+						};
+
+						// Provide matching suggestions
+						list_matches(last_word, options, ArraySize(options), false);
+					}
+					else if(conf_item->t == CONF_INT ||
+					        conf_item->t == CONF_UINT ||
+					        conf_item->t == CONF_UINT16 ||
+					        conf_item->t == CONF_LONG ||
+					        conf_item->t == CONF_DOUBLE ||
+					        conf_item->t == CONF_STRING ||
+					        conf_item->t == CONF_STRING_ALLOCATED ||
+						conf_item->t == CONF_JSON_STRING_ARRAY)
+					{
+						// pihole-FTL --config ... <int/long/double/string>
+						// Provide the default value as suggestion
+						char *value = NULL;
+						cJSON *val = addJSONConfValue(conf_item->t, &conf_item->d);
+						if(val != NULL && (value = cJSON_PrintUnformatted(val)) != NULL)
+						{
+							// Add '' to the output if it is a string
+							if(conf_item->t == CONF_STRING ||
+							   conf_item->t == CONF_STRING_ALLOCATED ||
+							   conf_item->t == CONF_JSON_STRING_ARRAY)
+							{
+								// If the value is a string, we need to add quotes
+								if(value[0] != '\'')
+								{
+									char *tmp = malloc(strlen(value) + 3);
+									if(tmp != NULL)
+									{
+										sprintf(tmp, "'%s'", value);
+										free(value);
+										value = tmp;
+									}
+								}
+							}
+
+							// If the default value starts with the last word we are trying to complete,
+							// print it as a suggestion
+							// If the last word is empty, print the value anyway
+							if(strStartsWith(value, last_word) || strlen(last_word) == 0)
+								puts(value);
+							free(value);
+						}
+						cJSON_Delete(val);
+					}
+					else if(conf_item->t == CONF_ENUM_PTR_TYPE)
+					{
+						// pihole-FTL --config dns.piholePTR ...
+						const char *options[] = {
+							"pi.hole", "hostname", "hostnamefqdn", "none"
+						};
+
+						// Provide matching suggestions
+						list_matches(last_word, options, ArraySize(options), false);
+					}
+					else if(conf_item->t == CONF_ENUM_BUSY_TYPE)
+					{
+						// pihole-FTL --config dns.replyWhenBusy ...
+						const char *options[] = {
+							"block", "allow", "refuse", "drop"
+						};
+
+						// Provide matching suggestions
+						list_matches(last_word, options, ArraySize(options), false);
+					}
+					else if(conf_item->t == CONF_ENUM_BLOCKING_MODE)
+					{
+						// pihole-FTL --config dns.blocking.mode ...
+						const char *options[] = {
+							"IP", "NX", "NULL", "IP_NODATA_AAAA", "NODATA"
```

Comment:
Does it make sense to list all value options here (as well in the other config keys) instead of getting them from the api source code where they are already defined? Like you did this with the theme?